lang/java/avro/src/main/java/org/apache/avro/generic/GenericDatumWriter.java

```diff
@@ -59,6 +60,7 @@ public GenericDatumWriter(Schema root, GenericData data) {
   public void setSchema(Schema root) { this.root = root; }
 
   public void write(D datum, Encoder out) throws IOException {
+    Objects.requireNonNull(out, "Encoder cannot be null");
```

Comment:
if it is really required, shouldn't the validation better be done in the called method (the protected one)?

Reply:
Oh I missed the access reasoning, thanks for clarifying. LGTM then.